components/brave_rewards/browser/rewards_notifications_service.h

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_REWARDS_BROWSER_REWARDS_NOTIFICATIONS_SERVICE_
+#define BRAVE_COMPONENTS_BRAVE_REWARDS_BROWSER_REWARDS_NOTIFICATIONS_SERVICE_
+
+#include <memory>
+
+#include "base/macros.h"
+#include "base/observer_list.h"
+#include "components/keyed_service/core/keyed_service.h"
+
+namespace brave_rewards {
+
+class RewardsNotificationsServiceObserver;
+
+class RewardsNotificationsService : public KeyedService {
+ public:
+  RewardsNotificationsService();
+  ~RewardsNotificationsService() override;
+
+  typedef uint32_t RewardsNotificationID;
+  typedef int64_t RewardsNotificationTimestamp;
+  typedef std::vector<std::string> RewardsNotificationArgs;
+
+  enum RewardsNotificationType {
+    REWARDS_NOTIFICATION_INVALID = 0,
+    REWARDS_NOTIFICATION_AUTO_CONTRIBUTE,
+    REWARDS_NOTIFICATION_GRANT,
+    REWARDS_NOTIFICATION_FAILED_CONTRIBUTION,
+    REWARDS_NOTIFICATION_IMPENDING_CONTRIBUTION,
+    REWARDS_NOTIFICATION_INSUFFICIENT_FUNDS,
+  };
+
+  struct RewardsNotification {
+    RewardsNotification()
```

Comment:
we need to add args to RewardsNotification sturct as well, so that we save args as well